storage/objects/main.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Sample objects creates, list, deletes objects and runs
+// other similar operations on them by using the Google Storage API.
+// More documentation is available at
+// https://cloud.google.com/storage/docs/json_api/v1/.
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"cloud.google.com/go/storage"
+)
+
+var objectName = fmt.Sprintf("golang-example-objects-%d", time.Now().Unix())
+
+const (
+	bucketName    = "golang-example-objects-bucket"
+	dstBucketName = "golang-example-objects-bucket-dst"
+)
+
+func main() {
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	if projectID == "" {
+		fmt.Fprintf(os.Stderr, "GOOGLE_CLOUD_PROJECT environment variable must be set.\n")
+		os.Exit(1)
+	}
+
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := createBucketIfNotexits(ctx, client, projectID, bucketName); err != nil {
+		log.Fatal(err)
+	}
+	// create another bucket later to be used to copy objects from the prev one.
+	if err := createBucketIfNotexits(ctx, client, projectID, dstBucketName); err != nil {
+		log.Fatal(err)
+	}
+	name := fmt.Sprintf("gs://%v/%v", bucketName, objectName)
+
+	if err := write(client, bucketName, objectName); err != nil {
+		log.Fatalf("failed to write to the object (%v): %v", objectName, err)
+	}
+
+	data, err := read(client, bucketName, objectName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("contents of %v: %q\n", name, string(data))
+
+	attrs, err := attrs(client, bucketName, objectName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("object attributes of %v: %+v\n", name, attrs)
+
+	err = makePublic(client, bucketName, objectName)
+	if err != nil {
+		log.Fatalf("failed to make the object public: %v", err)
```

Comment:
fmt for humans, log for machines. I understand why we should capitalize fmts but not sure about logs. Doing them both.